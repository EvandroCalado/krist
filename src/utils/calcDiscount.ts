export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
