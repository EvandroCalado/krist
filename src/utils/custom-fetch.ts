import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
