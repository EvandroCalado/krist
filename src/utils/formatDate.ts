export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
