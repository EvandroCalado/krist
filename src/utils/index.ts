export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
