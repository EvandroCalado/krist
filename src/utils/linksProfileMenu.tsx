import { Heart, MapPin, Package, Settings, UserRound } from 'lucide-react';

export const linksProfileMenu = [
  {
    id: 1,
    label: 'informações pessoais',
    path: 'profile',
    icon: <UserRound size={16} />,
  },
  {
    id: 2,
    label: 'meus pedidos',
    path: 'orders',
    icon: <Package size={16} />,
  },
  {
    id: 3,
    label: 'lista de desejos',
    path: 'wishlist',
    icon: <Heart size={16} />,
  },
  {
    id: 4,
    label: 'meus endereços',
    path: 'addresses',
    icon: <MapPin size={16} />,
  },
  {
    id: 5,
    label: 'configurar',
    path: 'settings',
    icon: <Settings size={16} />,
  },
];
