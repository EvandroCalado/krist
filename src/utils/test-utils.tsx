import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
