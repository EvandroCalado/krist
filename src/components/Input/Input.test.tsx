import { fireEvent, screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { Input } from '.';

describe('Input', () => {
  it('should render default', () => {
    customRender(<Input type="text" label="username" name="username" />);

    const label = screen.getByText('username');
    const input = screen.getByRole('textbox');

    expect(label).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue('');
    expect(input).toHaveAttribute('type', 'text');
    expect(input).toHaveAttribute('name', 'username');
  });

  it('should render type', () => {
    customRender(<Input type="password" label="password" name="password" />);

    const input = screen.getByLabelText('password');

    expect(input).toHaveAttribute('type', 'password');
  });

  it('should render on change', () => {
    customRender(<Input type="text" label="username" name="username" />);

    const input = screen.getByRole('textbox');

    expect(input).toHaveValue('');

    fireEvent.change(input, { target: { value: 'hello' } });

    expect(input).toHaveValue('hello');
  });

  it('should render to match snapshot', () => {
    const { container } = customRender(
      <Input type="text" label="text" name="text" />,
    );

    expect(container).toMatchSnapshot();
  });
});
