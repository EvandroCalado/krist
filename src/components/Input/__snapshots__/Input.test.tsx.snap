// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render to match snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: .8rem;
}

.c1 {
  font-size: 1.2rem;
  text-transform: capitalize;
}

.c2 {
  outline: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  border-radius: .8rem;
  border: 1px solid #131118;
}

.c2:focus {
  outline: 3px solid #131118;
  outline-offset: 1px;
}

.c2:disabled {
  opacity: 0.3;
  pointer-events: none;
}

<div>
  <div
    class="c0 input"
  >
    <label
      class="c1"
      for="text"
    >
      text
    </label>
    <input
      class="c2"
      id="text"
      name="text"
      type="text"
    />
  </div>
  ,
</div>
`;
